src/__experimental__/components/select/option.component.js

```diff
@@ -13,7 +13,7 @@ Option.propTypes = {
   /** Optional: alternative rendered content, displayed within <SelectList> of <Select> (eg: an icon, an image, etc) */
   children: PropTypes.node,
   /** The option's invisible internal value */
-  value: PropTypes.string.isRequired,
+  value: PropTypes.any.isRequired,
```

Comment:
I think that we should avoid using `any` type, you could use `oneOfType`, e.g.: `PropTypes.oneOfType([PropTypes.object, PropTypes.string])`

Reply:
Ok, just fixed that